### ISSUE TITLE
Dependencies need to be agreed upon

### ISSUE THREAD

**Issue by rcjackson** [ANL-DIGR/ACT#3]
Currently, ACT has the following dependencies:

* Numpy
* Xarray
* Matplotlib
* Astral (for shading in day vs night in timeseries plots)

as well as the packages these depend on such as scipy and pandas. We should agree about what dependencies ACT should have.

**Comment by AdamTheisen** [comment]
We could always look and see if something better could take the place of astral.  That's just what was used in the DQ Office for the shading.

We might also want to think about cartopy as well as I can see some contour plotting of data at SGP to be useful.

**Comment by AdamTheisen** [comment]
Just thinking, would there be a reason to have py-art as a dependency?

**Comment by scollis** [comment]
I would add it later when we flesh out some more stuff on the radars. I think the radar mentors will be a later adopter of this this technology.

**Comment by kenkehoe** [comment]
If you want to use xr.open_mfdataset to read more than one file at a time or to use the parallel feature you will also need dask.

**Comment by rcjackson** [comment]
We apparently need to constrain the minimum version of dask. Jenni ran into an issue where after installing it didn't work because it was using an old version of dask that doesn't implement delayed objects.

**Comment by kenkehoe** [comment]
OK do you know what version we need?

**Comment by rcjackson** [comment]
Let's make the minimum be the latest that is out, 1.1.5

________________________________

**Comment by rcjackson** [comment]
I have made MetPy an optional dependency for Skew-T plots, since the rest of ACT will work fine without MetPy. I figure this is the way to go unless we wind up having much more of ACT depend on MetPy.

**Comment by AdamTheisen** [comment]
Current list of dependencies

pandas
matplotlib
numpy
scipy
xarray
astral
dask
distributed
pint
cartopy
boto3
pyproj > '2.0.0'

Do we want to keep this open to track future dependencies, or can we close out and discuss future ones in the pull requests?

**Comment by kenkehoe** [comment]
How do we handle dependencies. I see one with a version number. I had to manually update cartopy above a version number to get it to work. Should we pay more attention on version number minimums?

**Comment by AdamTheisen** [comment]
@kenkehoe yes, if we have a minimum version, we should have that indicated. Do you remember what version it was?
